### PR TITLE
chore: move a few more files to TS

### PR DESCRIPTION
**Pull request by wegry**
Moves some more files over to TS from Flow (syntactically), `platform-implementation` and `compose-view` being the biggest two. I suggest ignoring whitespace changes. To reduce noise, _there should be no runtime changes_. If you see any, please flag them.

Most ESLint rules have been disabled where they conflict for newly moved Flow to TS files for the sake of noise. ~One exception is `prefer-const` autofixed in a few spots.~ _Moved to #916_. A subsequent pass to remove file wide `@typescript-eslint/no-unused-variables` disabling might flesh out a good chunk of dead code in the form of unused imports. 

Other bits:

- `Kefir.later(n)` has been rendered as `Kefir.later(n, undefined)` to match the TS Kefir types.
- `flow check` has been removed
- ~`.js.flow` files have been removed. This is roughly 1/3 of the files touched in the PR.~ _Moved to #917_
- most `getEventStream` methods on views tend to be loosely typed and should be tightened over time.

### CODE REVIEW

File: gulpfile.babel.ts [hunk 1]
```diff
@@ -188,7 +188,7 @@ async function webpackTask({
                 insert: (htmlElement: HTMLStyleElement) => {
                   htmlElement.setAttribute(
                     'data-inboxsdk-version',
-                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment -- this is injected by webpack
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment -- webpack provides SDK_VERSION at runtime
```
Comment (wegry) [inline gulpfile.babel.ts:191]:
```suggestion
// eslint-disable-next-line @typescript-eslint/ban-ts-comment -- this is injected by webpack
```

File: src/platform-implementation-js/constants/router.ts [hunk 2]
```diff
@@ -20,16 +18,27 @@ export const NATIVE_ROUTE_IDS: { [k: string]: string } = Object.freeze({
   CONTACTS: 'contacts/:page',
   CONTACT: 'contacts/:contactID',
   SETTINGS: 'settings/:section',
-  SCHEDULED: 'scheduled' /** DEC 2022 */,
```
Comment (Macil) [inline src/platform-implementation-js/constants/router.ts:23]: The comments here have all shifted to the wrong lines

File: src/platform-implementation-js/dom-driver/gmail/gmail-driver/user-info.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ export default class UserInfo {
     return this._userEmail;
   }
 
-  // deprecated
+  /* @deprecated */
```
[the remark this answers is not in the document]
Comment (wegry) [reply src/platform-implementation-js/dom-driver/gmail/gmail-driver/user-info.ts:33]: It does indeed.

```suggestion
  /** @deprecated */
```

**Review by Macil — APPROVED** [review]

File: src/injected-js/setup-event-reemitter.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            getAsString() {
+              throw new Error('getAsString not supported');
+            },
+          })),
+          types: ['Files'],
+
+          getData() {
+            return '';
+          },
+
```
Comment (wegry) [inline src/injected-js/setup-event-reemitter.ts:54]:
```suggestion
```

File: src/injected-js/setup-event-reemitter.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            getAsFile() {
+              return files[i];
+            },
+
+            getAsString() {
+              throw new Error('getAsString not supported');
+            },
+          })),
+          types: ['Files'],
+
```
Comment (wegry) [inline src/injected-js/setup-event-reemitter.ts:50]:
```suggestion
```

File: src/injected-js/setup-event-reemitter.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+          effectAllowed: 'all',
+          files,
+          items: files.map(({ type }: any, i: number) => ({
+            kind: 'file',
+            type,
+
+            getAsFile() {
+              return files[i];
+            },
+
```
Comment (wegry) [inline src/injected-js/setup-event-reemitter.ts:44]:
```suggestion
```

File: src/injected-js/setup-event-reemitter.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        }
+
+        (newEvent as any).dataTransfer = {
+          dropEffect: 'none',
+          effectAllowed: 'all',
+          files,
+          items: files.map(({ type }: any, i: number) => ({
+            kind: 'file',
+            type,
+
```
Comment (wegry) [inline src/injected-js/setup-event-reemitter.ts:40]:
```suggestion
```

File: src/injected-js/setup-event-reemitter.ts [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+          getData() {
+            return '';
+          },
+
+          setData() {},
+
+          setDragImage() {},
+        };
+      }
+
```
Comment (wegry) [inline src/injected-js/setup-event-reemitter.ts:60]:
```suggestion
```

File: src/injected-js/setup-event-reemitter.ts [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    'inboxsdk_event_relay',
+    function (event: Record<string, any>) {
+      const newEvent = document.createEvent('Events');
+      (newEvent as any).initEvent(
+        event.detail.type,
+        event.detail.bubbles,
+        event.detail.cancelable
+      );
+      Object.assign(newEvent, event.detail.props);
+
```
Comment (wegry) [inline src/injected-js/setup-event-reemitter.ts:19]:
```suggestion
```

File: src/injected-js/setup-event-reemitter.ts [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+      if (event.detail.dataTransfer) {
+        const { files, fileNames } = event.detail.dataTransfer;
+
```
Comment (wegry) [inline src/injected-js/setup-event-reemitter.ts:22]:
```suggestion
```

File: src/injected-js/setup-event-reemitter.ts [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+        if (fileNames) {
+          fileNames.forEach((fileName: string, i: number) => {
+            const file = files[i];
+
```
Comment (wegry) [inline src/injected-js/setup-event-reemitter.ts:26]:
```suggestion
```

File: src/injected-js/setup-event-reemitter.ts [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+            if (typeof file.name !== 'string') {
+              file.name = fileName;
+            }
+          });
+        }
+
```
Comment (wegry) [inline src/injected-js/setup-event-reemitter.ts:32]:
```suggestion
```

File: src/injected-js/setup-event-reemitter.ts [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+            },
+          })),
+          types: ['Files'],
+
+          getData() {
+            return '';
+          },
+
+          setData() {},
+
```
Comment (wegry) [inline src/injected-js/setup-event-reemitter.ts:56]:
```suggestion
```

**Review by wegry — COMMENTED** [review]
There may be a way to make this file show up as a minimal rename with some whitespace changes applied.

**Review by dmy7r0 — APPROVED** [review]
Went over the changes offline. Looks good

**Pull request merged**